src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -9502,5 +10167,784 @@ .maxstack  3
+    {
+        _ = new System.Action(this.M01);
+        _ = new System.Action(this.M04);
+    }
+
+    static void M04() {}
+
+    protected abstract static void M05();
+}
+
+class Test
+{
+    static void MT1(I1 x)
+    {
+        _ = new System.Action(I1.M01);
+        _ = new System.Action(x.M01);
+        _ = new System.Action(I1.M04);
+        _ = new System.Action(x.M04);
+    }
+
+    static void MT2<T>() where T : I1
+    {
+        _ = new System.Action(T.M03);
+        _ = new System.Action(T.M04);
+        _ = new System.Action(T.M00);
+        _ = new System.Action(T.M05);
+
+        _ = (System.Linq.Expressions.Expression<System.Action>)(() => new System.Action(T.M01).ToString());
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            compilation1.VerifyDiagnostics(
+                // (8,31): error CS9107: A static abstract interface member can be accessed only on a type parameter.
+                //         _ = new System.Action(M01);
+                Diagnostic(ErrorCode.ERR_BadAbstractStaticMemberAccess, "M01").WithLocation(8, 31),
+                // (14,31): error CS0176: Member 'I1.M01()' cannot be accessed with an instance reference; qualify it with a type name instead
+                //         _ = new System.Action(this.M01);
+                Diagnostic(ErrorCode.ERR_ObjectProhibited, "this.M01").WithArguments("I1.M01()").WithLocation(14, 31),
+                // (15,31): error CS0176: Member 'I1.M04()' cannot be accessed with an instance reference; qualify it with a type name instead
+                //         _ = new System.Action(this.M04);
+                Diagnostic(ErrorCode.ERR_ObjectProhibited, "this.M04").WithArguments("I1.M04()").WithLocation(15, 31),
+                // (27,31): error CS9107: A static abstract interface member can be accessed only on a type parameter.
+                //         _ = new System.Action(I1.M01);
+                Diagnostic(ErrorCode.ERR_BadAbstractStaticMemberAccess, "I1.M01").WithLocation(27, 31),
+                // (28,31): error CS0176: Member 'I1.M01()' cannot be accessed with an instance reference; qualify it with a type name instead
+                //         _ = new System.Action(x.M01);
+                Diagnostic(ErrorCode.ERR_ObjectProhibited, "x.M01").WithArguments("I1.M01()").WithLocation(28, 31),
+                // (30,31): error CS0176: Member 'I1.M04()' cannot be accessed with an instance reference; qualify it with a type name instead
+                //         _ = new System.Action(x.M04);
+                Diagnostic(ErrorCode.ERR_ObjectProhibited, "x.M04").WithArguments("I1.M04()").WithLocation(30, 31),
+                // (35,31): error CS0119: 'T' is a type parameter, which is not valid in the given context
```

Comment:
not for this PR, but I feel like we should look at changing this diagnostic message to indicate that the reason `T` can't be used as a receiver is because `M03` is not `static abstract`.